api/src/main/java/io/jsonwebtoken/SignatureAlgorithm.java

```diff
@@ -574,8 +581,16 @@ public static SignatureAlgorithm forSigningKey(Key key) throws InvalidKeyExcepti
         if (key instanceof SecretKey) {
 
             SecretKey secretKey = (SecretKey)key;
+            String secretKeyAlg = secretKey.getAlgorithm();
             int bitLength = io.jsonwebtoken.lang.Arrays.length(secretKey.getEncoded()) * Byte.SIZE;
 
+            // first, check the key alg name
+            SignatureAlgorithm algFromJcaName = PREFERRED_HMAC_ALGS_BY_JCA.get(secretKeyAlg);
```

Comment:
A couple of things here:

1.  The algorithm names, as we saw on Android, can be case-insensitive.  If we're going to do a map lookup, the map keys should be lowercased (or uppercased) before map insertion to ensure a guaranteed lookup.  Additionally `secretKeyAlg` should be similarly lowercased (or uppercased) - whatever the map keys are - to ensure a valid lookup.

2.  It's not just standard JCA names that we have to worry about.  An OID String value - instead of a name - could also be returned on some JVMs, e.g. elsewhere in `SignatureAlgorithm` we have this logic:
```java
// These next checks use equalsIgnoreCase per https://github.com/jwtk/jjwt/issues/381#issuecomment-412912272
if (!HS256.jcaName.equalsIgnoreCase(alg) &&
    !HS384.jcaName.equalsIgnoreCase(alg) &&
    !HS512.jcaName.equalsIgnoreCase(alg) &&
    !HS256.pkcs12Name.equals(alg) &&
    !HS384.pkcs12Name.equals(alg) &&
    !HS512.pkcs12Name.equals(alg)) {
        throw new InvalidKeyException("The " + keyType(signing) + " key's algorithm '" + alg +
        "' does not equal a valid HmacSHA* algorithm name and cannot be used with " + name() + ".");
    }
}
```
These OID values would also have to be keys in the lookup map to ensure proper functionality.

3.  Line 585 above, `int bitLength = io.jsonwebtoken.lang.Arrays.length(secretKey.getEncoded()) * Byte.SIZE;` could throw an exception on JVMs (e.g. Hardware Security Modules) or later versions of Android because the codebase doesn't have access to the raw key bytes.  In these scenarios (`getEncoded()` throws an exception *AND* `getAlgorithm()` returns a valid identifier in the map), we should return the key as valid because there's no way for JJWT to perform a bit-strength check.  I'm wondering if we need to add this logic now however - it could be added in another PR. Probably should to isolate the two PRs in case there are any issues that arise out of it.

4. The `jwe` branch has deprecated the `SignatureAlgorithm` enum in favor of a new `SignatureAlgorithms` class because an enum doesn't allow pluggable signature algorithm implementations.  Any change in this PR needs to ensure the identical functionality exists in the `jwe` branch's `SignatureAlgorithms` class.  Can you please investigate that and open a PR there referencing the same ticket # to ensure it's captured appropriately?